### ISSUE TITLE
Allow .zip downloads as well.

### ISSUE THREAD

**Issue by jelaw21** [SKCraft/Launcher#331]
I have read the previous post about including a .zip of the modpack and the reasoning behind not implementing it. However, Curse and Twitch have changed the playing a field a bit and now modpacks are All rights reserved, AND the mods inside them. 

I attempted this with my own modpack:
`Skyblock+Billionaire-1.5.7.zip`
`Skyblock+Billionaire-1.5.7.zip.url.txt`

In Test it worked, and I could verify that the number of downloads for my alpha pack went up. 

However, when I went to Build I got this error:
[warning] Task failed
java.lang.NullPointerException
	at java.io.File.<init>(Unknown Source)
	at com.skcraft.launcher.builder.ClientFileCollector.onFile(ClientFileCollector.java:76)
	at com.skcraft.launcher.builder.DirectoryWalker.walk(DirectoryWalker.java:73)
	at com.skcraft.launcher.builder.DirectoryWalker.walk(DirectoryWalker.java:43)
	at com.skcraft.launcher.builder.PackageBuilder.addFiles(PackageBuilder.java:114)
	at com.skcraft.launcher.builder.PackageBuilder.main(PackageBuilder.java:392)
	at com.skcraft.launcher.creator.controller.task.PackBuilder.call(PackBuilder.java:66)
	at com.skcraft.launcher.creator.controller.task.PackBuilder.call(PackBuilder.java:21)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Anyway of a fix? I would love to credit modpack creators with downloads along with crediting the mod authors.

**Comment by spannerman79** [comment]
>Curse and Twitch have changed the playing a field a bit and now modpacks are All rights reserved, AND the mods inside them.

Curse/Twitch cannot all rights reserved a mod if its legally open sourced or licenced as GPL/MIT/etc.  Its actually illegal.

For example, lets take [AE2](https://www.curseforge.com/minecraft/mc-mods/applied-energistics-2) which licence is https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/rv6-1.12/LICENSE 

It would be illegal for Curse/Twitch to change that licence without consent from all code contributors.  So whatever Curse/Twitch is stating is utter BS and/or off their rocker smoking something that they shouldn't be

>However, when I went to Build I got this error:

See #235

**Comment by jelaw21** [comment]
I've read through that and I'm not entirely sure what my fix would be, the new url's mentioned at the top of the post? 

Also, maybe not the mods, but the modpacks. For example, was told by an FTB representative that if I host an FTB pack publicly, that's illegal. And since I'm making a launcher to release on our discord, I'd say that's public, yea? However, if I include a download to the pack, (so they get their download number) that would be acceptable. This is all I'm trying to achieve.

**Comment by spannerman79** [comment]
FTB representative ≠ Twitch/Curse See [here](https://twitter.com/FTB_Team/status/1181606400159633408)

>For example, was told by an FTB representative that if I host an FTB pack publicly, that's illegal

A direct copy/paste of one - yes it is. Take a read of [this](https://www.reddit.com/r/feedthebeast/comments/6ploeu/mod_pack_copyright/dkqdhqa/) first before you read on further.

If you do **major** config changes to suit your server, changing recipe (look at GT:NH for one example for major recipe changes) outside the original scope that mght fall under fair use (excluding graphic artwork) then yes you can.

If FTB was still partnered with Twitch you _might_ have gotten away with it if you were a Linux user (there is no _native_ Twitch App for Linux) but now that FTB have cut the cord they have fallen back to their original java launcher.

In short;
This Launcher is designed around supporting a modpack that **you** have made.  Not to host others.

**Comment by jelaw21** [comment]
Yes. I realize the intent of the Launcher and know for a fact of two
communities using it to host FTB packs and others.

I want to give credit to the authors. Without that, progress and
development slows. You have provided a way for mod authors to get credit
via the url.txt files. I'm trying to figure out how to obtain the modpack
files in much the same way. Whether it's those exact files that get sent to
the user or not, I think copyright just states that the packs have to
downloaded from Curse/Forge. So I was attempting to use the url.txt
technique to download a modpack zip folder so the author gets credit.

Something similar to this: https://github.com/Franckyi/CMPDL . But really
just a way to download that .zip. Not to extract it or anything, as this
files are already in the build. Just to ping a download count for the
author.

**Comment by spannerman79** [comment]
> You have provided a way for mod authors to get credit
via the url.txt files. I'm trying to figure out how to obtain the modpack
files in much the same way.

No, the file formats for pack management are drastically different

> I think copyright just states that the packs have to
downloaded from Curse/Forge.

Again no. You obviously need to stop what you are doing as you clearly do not understand copyright.

**Comment by hedgehog1029** [comment]
Closing this as out-of-scope.
